lib/spark/options/docs.ex

```diff
@@ -435,17 +444,73 @@ defmodule Spark.Options.Docs do
           [type1, type2] -> quote(do: {unquote(type1), unquote(type2)})
           types -> quote do: {unquote_splicing(types)}
         end
+
+      {:tagged_tuple, tag, subtype} ->
+        quote do: {unquote(tag), type_to_spec(unquote(subtype))}
+
+      {tag, _} when tag in [:behaviour, :spark_behaviour, :protocol, :spark] ->
+        quote(do: module())
+
+      {:spark_function_behaviour, _, _} ->
+        quote(do: module())
+
+      {:spark_function_behaviour, _, _, _} ->
+        quote(do: module())
+
+      {:spark_type, _, _} ->
+        quote(do: module)
+
+      {:spark_type, _, _, _} ->
+        quote(do: module)
+
+      {:mfa_or_fun, arity} ->
+        quote do: type_to_spec(:mfa) | type_to_spec(:fun, unquote(arity))
```

Comment:
This seems weird, `type_to_spec` won't be available when this is unquoted, right? I think we'd need to `unquote` the whole call to `type_to_spec/1`

Reply:
good call. maybe I should write some tests lol :)